### DISCUSSION TITLE
Credentials when using helper and docker

### DISCUSSION THREAD

**Discussion by wz2b** [zachowj/node-red-contrib-home-assistant-websocket#569] [Q&A]
When you click "I use the Home Assistant Add-on" it finds a credentials type key somewhere, but my node-red isn't finding it.  I have the helper integration installed into home assistant and it isn't giving any errors so I'm pretty sure it's working.  But, home assistant is running in docker, and node-red is not.

How does the node-red add-on find the credentials?  Is it trying to access it through ~homeassistant/.homeassistant somewhere?  Is there any issue with this working correctly if home-assistant is running in docker, but node-red is external?

**Comment by zachowj** [comment]
Are you running NR as a stand-alone docker instance or are you using the Home Assistant Add-on? <https://github.com/hassio-addons/addon-node-red>

**Reply by wz2b** [reply]
I'm running NR standalone on the machine (not in a container) and home-assistant is running in docker.  Because home-assistant is in docker and not using the home assistant supervisor I can't install add-ons through the UI.

Maybe what I did was wrong, then, because what I installed wasn't an "add-on" it was a "companion integration."  I installed that by following these instructions:

    https://github.com/zachowj/hass-node-red

Maybe that's not the same thing?

**Reply by zachowj** [reply]
In the server config in NR you do not want to check the "I use the Home Assistant Add-on" because you are not using the add-on.

<https://zachowj.github.io/node-red-contrib-home-assistant-websocket/guide/#configuration>

**Comment by wz2b** [comment]
Doesn't that mean I can't use a whole bunch of nodes, though?

**Reply by wz2b** [reply]
https://github.com/hassio-addons/addon-node-red/blob/main/node-red/DOCS.md 

Where it says "Example add-on configuration" where does that file live?
